contracts/test/LightClient.t.sol

```diff
@@ -112,6 +142,124 @@ contract LightClient_constructor_Test is LightClientCommonTest {
     }
 }
 
+contract LightClient_permissionedProver_Test is LightClientCommonTest {
+    LC.LightClientState internal newState;
+    V.PlonkProof internal newProof;
+
+    function setUp() public {
+        init();
+
+        string[] memory cmds = new string[](6);
+        cmds[0] = "diff-test";
+        cmds[1] = "mock-consecutive-finalized-states";
+        cmds[2] = vm.toString(BLOCKS_PER_EPOCH_TEST);
+        cmds[3] = vm.toString(STAKE_TABLE_CAPACITY / 2);
+        cmds[4] = vm.toString(uint64(1));
+        cmds[5] = vm.toString(uint64(1));
+
+        bytes memory result = vm.ffi(cmds);
+        (LC.LightClientState[] memory states, V.PlonkProof[] memory proofs) =
+            abi.decode(result, (LC.LightClientState[], V.PlonkProof[]));
+
+        newState = states[0];
+        newProof = proofs[0];
+    }
+
+    function test_NoProverPermissionsRequired() external {
+        //set permissioned flag to false
+        vm.expectEmit(true, true, true, true);
+        emit LC.PermissionedProverModeDisabled();
+        vm.prank(admin);
+        lc.setPermissionedProverMode(false);
+
+        //any prover can call the newFinalizedState method as the contract is not in permissioned
+        // prover mode
+        vm.prank(makeAddr("randomUser"));
+        lc.newFinalizedState(newState, newProof);
+    }
+
+    function test_NoProverPermissionsRequiredAndProverNotSet() external {
+        //set permissioned flag to false
+        vm.expectEmit(true, true, true, true);
+        emit LC.PermissionedProverModeDisabled();
+        vm.prank(admin);
+        lc.setPermissionedProverMode(false);
+
+        //assert that the contract is not permissioned
+        assert(lc.permissionedProverMode() == false);
+
+        //any prover can call the newFinalizedState method as the contract is not in permissioned
+        // prover mode
+        vm.prank(makeAddr("randomUser"));
+        lc.newFinalizedState(newState, newProof);
+    }
+
+    function test_NoProverPermissionsRequiredAndNoProverSet() external {
+        //deploy a fresh light client proxy
+        (address newLcTestProxy,,) = deployer.deployContract(genesis, BLOCKS_PER_EPOCH_TEST, admin);
+
+        //cast the proxy to be of type light client test
+        LCTest newLc = LCTest(newLcTestProxy);
+
+        //assert that the contract is not permissioned
+        assert(newLc.permissionedProverMode() == false);
+        //assert that the permissionedProver has not been set and thus the zero address
+        assert(newLc.permissionedProver() == address(0));
+
+        //any prover can call the newFinalizedState method as the contract is not in permissioned
+        // prover mode
+        vm.prank(makeAddr("randomUser"));
+        newLc.newFinalizedState(newState, newProof);
+    }
+
+    function test_RevertWhen_NoPermissionedProverSetInPermissionedMode() external {
```

Comment:
Maybe this has been discussed and I missed it but I wonder what the use case is for being able to enable permissioned prover mode without having a permissioned prover set. We could also make this impossible at the contract level.

Reply:
In other words we could have a function to disable the permissioned mode and one to enable it that takes an address as an argument that can't be the zero address. Then it's no longer possible to "disable" the lightlclient contract. Currently it seems it's possible to set permissioned prover mode without a permissioned prover which in fact disables sending proofs. If having a disabled state is desired we should make it explict, if it's not desired I think it's best to make it less likely. It's of course still possible to set an address that doesn't exist.

cc @alysiahuggins @philippecamacho

Reply:
i agree that you should not be able to update the permissioned prover mode in an instance with the prover address is not specified. I'll update the function.

Reply:
addressed here https://github.com/EspressoSystems/espresso-sequencer/commit/c6ff57f7d67653b9bc5bb3bf69bda7f5609f779e

Reply:
I think setting the prover to zero address if permissioned mode is disabled is a good idea. I think the `update` function is fine, for the other one, I think we should split it in two to get rid of the boolean parameter and make it more obvious to the operator what's they are doing. E. g.

1. `enablePermissionedProverMode(address)`
2. `disablePermissionedProverMode()`

What do you think?

Reply:
Yea that's clearer should reduce complexity for users, will do!

Reply:
refactored here https://github.com/EspressoSystems/espresso-sequencer/commit/830ab2f0a835ba46a897cb429b68f5ed0bb57c98